pyaccel/new_lifetime.py

```diff
@@ -107,20 +131,32 @@ def inelastic(accep, P):
 def Touschek(emit, acoplamento, Bx, By, eta_x, spread, accep, sig_S, Ib, T_rev, qe, r0, c, gamma):
 
     #Calcula Touschek
-    Dp = 1e-1
+    
     emitx = emit*(1-acoplamento)
     emity = emit*acoplamento
     V = np.zeros(len(Bx))
+    sigx = np.zeros(len(Bx))
+    Dp = np.zeros(len(Bx))
     for i in range (len(Bx)):
-        sigx = math.sqrt(emitx*Bx[i]+(eta_x[i]*spread)**2)
+        sigx[i] = math.sqrt(emitx*Bx[i]+(eta_x[i]*spread)**2)
         sigy = math.sqrt(emity*By[i])
-        V[i] = sig_S * sigx * sigy
+        V[i] = sig_S * sigx[i] * sigy
+        Dp[i] = Calc_D(accep, sigx[i], gamma)
+
     N = Ib * T_rev / qe 
     inv_T = (r0**2*c/8/math.pi)*N/gamma**2/ accep**3 * Dp / V
 
-    return (inv_T[0])**-1
+    return (inv_T[ring_loc])**-1
 
+def Calc_D(accep, sigx,gamma):
+    eps = (accep/(sigx*gamma))**2
 
+    f1 = lambda x : np.exp(-x)*np.log(x)/x
+    f2 = lambda x : np.exp(-x)/x
+    I1 = integrate.fixed_quad(f1, eps, 100*eps, n=1000)
```

Comment:
I believe that there are better alternatives in scipy to perform integration when the integration limit is infinity.